app/assets/javascripts/admin/products/controllers/units_controller.js.coffee

```diff
@@ -21,7 +21,13 @@ angular.module("admin.products")
           $scope.product.variant_unit = $scope.product.variant_unit_with_scale
           $scope.product.variant_unit_scale = null
       else
-        $scope.product.variant_unit = $scope.product.variant_unit_scale = null
+        $scope.product.variant_unit = $scope.product.variant_unit || angular.element('#product_variant_unit').val() || null
+        $scope.product.variant_unit_scale = $scope.product.variant_unit_scale || parseFloat(angular.element('#product_variant_unit_scale').val()) || null
+        $scope.product.variant_unit_scale = null if isNaN($scope.product.variant_unit_scale)
+        $scope.product.variant_unit_with_scale = $scope.product.variant_unit + '_' + $scope.product.variant_unit_scale if !!$scope.product.variant_unit
+        setTimeout(->
+          $('#product_variant_unit_with_scale').trigger('change')
+        , 1000)
```

Comment:
le hack to fix le bigger hack (select2 + angular)

Reply:
Can you describe in a little bit more details what this is doing and why we need it?

Reply:
Basically, we are mixing angular and select2 (a jquery lib for selects) and the two aren't realy designed to work together. To me that's the bigger hack here. We should be using something like angular UI so that's it's all speaking the same language. As it happens, select2 doesn't get updated when angular events happen so I'm forcing the event here.

Reply:
Pretty good explanation. IMO it deserves to be commented in the code as well, although summarized.